snuba/datasets/schemas/__init__.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC, abstractmethod
+from typing import Mapping, List
+
+from snuba import settings
+from snuba.clickhouse.columns import ColumnSet
+
+
+def local_dataset_mode():
+    return settings.DATASET_MODE == "local"
+
+
+class Schema(ABC):
+    """
+    A schema is an abstraction over a data model we can query.
+    It provides a set of columns and a where clause to build the query.
+    Concretely this can represent a table, a view or a group of
+    joined tables.
+    This level of abstraction only provides read primitives.
+
+    As of now this we do not have a strict separation between a Snuba abstract
+    schema and a Clickhouse concrete schema. When this will exist, this
+    class will break up into snuba schema and clickhouse schema.
+    """
+
+    def __init__(
+        self,
+        *,
+        columns: ColumnSet,
+    ) -> None:
+        self.__columns = columns
+
+    @abstractmethod
+    def get_from_clause(self) -> str:
+        """
+        Builds and returns the content of the FROM clause we need
+        to pass to Clickhouse to execute a query on this schema.
```

Comment:
I think this should make a distinction that this doesn't literally contain the "`FROM `" part of the clause. I think it'd be great to include that in the method name itself but I have no idea how to communicate that. 😕 

Actually, I guess this will be less of an issue later, based on the TODO comment. I guess the `FROM` structure will also include the `JOIN` specification?

Reply:
> get_data_sources

That works, it looks like PostgreSQL refers to this as a `from_item` in the SQL description as well.

> Do you mean in the current implementation or in the future implementation? I think the answer is yes in both cases but in very different forms.

Yeah, I guess that makes sense. 🙂 I meant the future implementation (whatever non-string data structure actually represents the `FROM` clause.)